front/app/containers/Admin/inspirationHub/useCountryCodeSupportedInProjectLibrary.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+import { coreSettings } from 'api/app_configuration/utils';
+import useProjectLibraryCountries from 'api/project_library_countries/useProjectLibraryCountries';
+
+const useCountryCodeSupportedInProjectLibrary = () => {
+  const { data: appConfiguration } = useAppConfiguration();
+  const country_code = appConfiguration
+    ? coreSettings(appConfiguration.data).country_code
+    : null;
+
+  // If the project library countries fetch fails,
+  // react-query will retry the request indefinitely for some reason.
+  // This piece of state is used to prevent that.
+  const [fetchedWithError, setFetchedWithError] = useState(false);
+  const { data: projectLibraryCountries, error } = useProjectLibraryCountries({
+    enabled: !fetchedWithError,
+  });
+
+  useEffect(() => {
+    if (fetchedWithError) return;
+    if (error) {
+      setFetchedWithError(true);
+    }
+  }, [error, fetchedWithError]);
+
+  if (fetchedWithError) {
+    return { status: 'error', countryCode: null } as const;
+  }
+
+  if (!projectLibraryCountries || !country_code) {
+    return { status: 'loading', countryCode: null } as const;
+  }
+
+  const countryCodeSupported = (() => {
+    const countriesSupportedByLibrary =
+      projectLibraryCountries.data.attributes.map((country) => country.code);
+
+    return countriesSupportedByLibrary.includes(country_code);
+  })();
+
+  if (!countryCodeSupported) {
+    return { status: 'not-supported', countryCode: null } as const;
+  }
```

Comment:
```suggestion
  const countriesSupportedByLibrary =
    projectLibraryCountries.data.attributes.map((country) => country.code);

  if (!countriesSupportedByLibrary.includes(country_code)) {
    return { status: 'not-supported', countryCode: null } as const;
  }
```

Easier to understand? :)

Reply:
You could also keep the `countryCodeSupported` variable if you think that makes it even more readable, but without the function call. Just the bit where it checks if the country code is in the array.